src/api/callback.cc

```diff
@@ -101,32 +101,20 @@ void InternalCallbackScope::Close() {
     return;
   }
 
-  TickInfo* tick_info = env_->tick_info();
-
   if (!env_->can_call_into_js()) return;
 
   auto weakref_cleanup = OnScopeLeave([&]() { env_->RunWeakRefCleanup(); });
 
-  if (!tick_info->has_tick_scheduled()) {
-    MicrotasksScope::PerformCheckpoint(env_->isolate());
-  }
-
   // Make sure the stack unwound properly. If there are nested MakeCallback's
   // then it should return early and not reach this code.
   if (env_->async_hooks()->fields()[AsyncHooks::kTotals]) {
     CHECK_EQ(env_->execution_async_id(), 0);
     CHECK_EQ(env_->trigger_async_id(), 0);
   }
 
-  if (!tick_info->has_tick_scheduled() && !tick_info->has_rejection_to_warn()) {
-    return;
-  }
-
   HandleScope handle_scope(env_->isolate());
   Local<Object> process = env_->process_object();
 
-  if (!env_->can_call_into_js()) return;
```

Comment:
Please keep this line :)

Reply:
There's one above (line 104), I don't think we do anything that would change the state of it between there and here now? (Prior to that it was the microtask queue.)

Reply:
Ah right, makes sense :+1: